src/types-and-traits.rst

```diff
@@ -1637,13 +1637,65 @@ Type Unification
 .. rubric:: Legality Rules
 
 :dp:`fls_ryvdhkgm7vzj`
-:t:`Type unification` is a measure of compatibility between two :t:`[type]s`. A
-:t:`type` is said to :t:`unify` with another :t:`type` when the domains,
-ranges, and structures of both :t:`[type]s` are compatible.
+:t:`Type unification` is the process by which :t:`type inference` propagates
+known :t:`[type]s` across the :t:`type inference root` and assigns concrete
+:t:`[type]s` to :t:`[type variable]s`, and also a general mechanism to check
+for compatibility between two :t:`[type]s` during :t:`method resolution`.
+
+:dp:`fls_67VZrx6dw68H`
+A :t:`type` is said to :t:`unify` with another :t:`type` when the domains,
+ranges, and structures of both :t:`[type]s` are compatible according to the
+rules detailed below.
 
 :dp:`fls_aie0tr62vhw5`
 Two types that :t:`unify` are said to be :t:`[unifiable type]s`.
 
+:dp:`fls_3U7Ue6Xzuv9M`
+:t:`Type unification` is a symmetric operation. If a :t:`type` ``A`` unifies
+with a :t:`type` ``B``, then ``B`` also unifies with ``A`` and such
+:t:`type unification` results in the same observable effects.
+
+:dp:`fls_tIiw5WkTRNf7`
+If one of the two :t:`[type]s` is a :t:`type variable`, :t:`type unification`
+proceeds as follows:
+
+* :dp:`fls_EoEbldkKBQW0`
```

Comment:
I suspect that these list items should be part of a numbered list as the logic implies ordering.

Reply:
Yeah, good point. Did that.